org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -20,10 +30,38 @@ public class NearInteractionModeDetector : ProximityDetector
         [Tooltip("The set of near interactors that belongs to near interaction")]
         private List<XRBaseInteractor> nearInteractors;
 
+        /// <summary>
+        /// Used to keep track of the previously detected colliders so that we can know which
+        /// colliders stopped being detected and update their buttons front plate and rounded
+        /// rect if they are labeled as dynamic (based on proximity).
+        /// </summary>
+        List<Collider> previouslyDetectedColliders = new List<Collider>();
+
         /// <inheritdoc />
         public override bool IsModeDetected()
         {
-            return base.IsModeDetected() || IsNearInteractorSelecting();
+            bool result = base.IsModeDetected() || IsNearInteractorSelecting();
+            if (result)
+            {
+                for (int i = 0; i < previouslyDetectedColliders.Count; i++)
+                {
+                    Collider previouslyDetectedCollider = previouslyDetectedColliders[i];
+                    if (!DetectedColliders.Contains(previouslyDetectedCollider) && previouslyDetectedCollider != null)
+                    {
+                        previouslyDetectedCollider.GetComponent<IPressableButtonForNearInteractionModeDetector>()?.UpdateFrontPlateAndRoundedRectIfDynamic(false);
+                        previouslyDetectedColliders.Remove(previouslyDetectedCollider);
+                    }
+                }
+                foreach (Collider collider in DetectedColliders)
+                {
+                    if (!previouslyDetectedColliders.Contains(collider))
```

Comment:
using a hash set for `previouslyDetectedColliders` will be more performant.   HashSet<Collider>

Reply:
Thank you for the feedback.  As we discussed offline, the collection gets modified by other instances, for example if I switch from a List to a Hashset as in:
![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/84108471/9d2812e1-ad17-4b4c-8ece-352e023801c4)
which I would traverse with:
![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/84108471/a9de0730-da8e-4ea1-950e-aeb902163215)
then I get several exceptions as this:
![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/84108471/da7a6567-4ca5-4ce7-a253-22488a4df36e)
I could make a copy of the Hashset to an array and traverse it with a for cycle but that would make it less performant and increase the memory consumption.
Because of the above, as we discussed offline, I will keep the use of a List.  I do appreciate your feedback and the time you invested in reviewing this PR though.  Resolving this conversation.